compiler/lib/source_map.mli

```diff
@@ -62,6 +62,13 @@ val empty : filename:string -> t
 
 val to_string : t -> string
 
-val to_file : t -> string -> unit
-
 val of_string : string -> t
+
+val of_file_no_mappings : string -> t * string option
```

Comment:
It seems that #1617 should fix the issue by delaying the parsing of mapping until necessary

Reply:
Yes. I considered not submitting these `Source_map` API changes, but it does make merging JSOO and WSOO a bit harder. If they are not acceptable as a temporary measure until #1617 is in proper shape, I will take them out and I will have to find another way to reconcile changes during the transitory period.